conda-store-server/conda_store_server/server/auth.py

```diff
@@ -58,18 +70,88 @@ def authenticate(self, token):
 
 
 class RBACAuthorizationBackend(LoggingConfigurable):
+    role_mappings_version = Integer(
+        1,
+        help="Role mappings version to use: 1 (default, legacy), 2 (new, recommended)",
+        config=True,
+    )
+
+    def _database_role_bindings_v1(self, entity: schema.AuthenticationToken):
+        with self.authentication_db() as db:
+            result = db.execute(
+                text(
+                    """
+                    SELECT nrm.entity, nrm.role
+                    FROM namespace n
+                    RIGHT JOIN namespace_role_mapping nrm ON nrm.namespace_id = n.id
+                    WHERE n.name = :primary_namespace
+                    """
+                ),
+                {"primary_namespace": entity.primary_namespace},
+            )
+            raw_role_mappings = result.mappings().all()
+
+            db_role_mappings = defaultdict(set)
+            for row in raw_role_mappings:
+                db_role_mappings[row["entity"]].add(row["role"])
+
+        return db_role_mappings
```

Comment:
This code used to be in `database_role_bindings`. I moved it to the start of the class to use it in `_role_mappings_versions`.